src/api-client/server/index.ts

```diff
@@ -120,4 +127,67 @@ export default {
 
     return `${process.env.VUE_APP_BACKEND_URL}/media/${md5(uri)}`;
   },
+  updateUser(token: string, user: string, updatedUser: User): Promise<void> {
+    // TODO: implements
+    return Promise.reject(
+      `not implemented yet. ${token}/${user}/${updatedUser}`
+    );
+  },
+  updatePassword(
+    token: string,
+    user: string,
+    oldPassword: string,
+    newPassword: string
+  ): Promise<void> {
+    // TODO: implements
+    return Promise.reject(
+      `not implemented yet. ${token}/${user}/${oldPassword}/${newPassword}`
+    );
+  },
+  // Fediverse
+  async fediverseSearchUserId(query: string): Promise<string | undefined> {
+    return await axios
+      .get<Webfinger>(
+        `https://${query.substring(
+          query.indexOf("@"),
+          query.length
+        )}/.well-known/webfinger?resource=acct:${query}`
+      )
+      .then((resp) => {
+        const webfinger = resp.data;
+        const link = webfinger.links.find(($) => $.rel === "self");
+        return link?.href;
+      });
+  },
+  async fediverseGetActor(url: string): Promise<Actor> {
+    return await axios
+      .get(url, {
+        headers: {
+          Accept: "application/activity+json",
+        },
+      })
+      .then((resp) => {
+        return resp.data;
+      });
+  },
+  async fediverseGetUser(url: string): Promise<User> {
```

Comment:
what's `User` and what's and `Actor`?

Reply:
I have distinguished Actor and User so that User is an extension of Actor. User also contains, for example, the URL for the followers and following lists, which, as you have seen, can also be called up and displayed in the search result

Reply:
But every actor should have a follower/following list according to the AP standard?

Reply:
You're right. You would have to add the actor interface in the library activitypub-objects. You would have to adapt these first.

Reply:
Maybe create an issue for it and postpone this?

Reply:
Issue was created: #71